examples/cloudml-churn-prediction/preprocessor/preprocessor/preprocess.py

```diff
@@ -0,0 +1,325 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Build preprocessing pipeline for survival analysis"""
+
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import PipelineOptions, WorkerOptions
+from tensorflow_transform.beam import impl as tft_beam
+from tensorflow_transform.beam import tft_beam_io
+from tensorflow_transform import coders
+import tensorflow_transform as tft
+from tensorflow_transform.tf_metadata import dataset_schema
+
+import logging
+import sys
+import argparse
+import time
+from datetime import datetime, timedelta
+import random
+import numpy as np
+import os
+import posixpath
+
+from . import query
+from . import features
+
+
+def random_date(start, end):
+    """Generate random date. 
+    Used to randomly generate (fake) subscription start_date
+    """
+    delta = end - start
+    int_delta = delta.days
+    random_day = random.randrange(int_delta)
+    return start + timedelta(days=random_day)
+
+
+def randomDuration(start):
+    """Generate random duration.
+    Used to random generate (fake) subscription duration, which directly relates
+    to the artificial label
+    """
+    random_day = random.randrange(np.max(features.LABEL_CEILINGS)
+        + np.min(features.LABEL_CEILINGS))
+    return start + timedelta(days=random_day), random_day
+
+
+def _generateFakeData(element):
+    """Appends randomly generated labels to each sample's dictionary.
+    It should not be used with real data.
+    Function does not add duration to the dictionary, because this field generally
+    must be calculated rather than directly output from a database.
+
+    Args:
+        element: dictionary of results from BigQuery
+    Returns:
+        dictionary of results from BigQuery with the following fields added:
+            start_date:
+            end_date: None if subscription has not yet ended
+            active:
+    """
+    d1 = datetime.strptime('1/1/2018', '%m/%d/%Y')
```

Comment:
I feel like these dates should be pulled into some sort of constants file or into `parse_args`. Someone with a new dataset they would want to apply to this template would likely want to edit the dates without digging through the code.

Reply:
These dates are just used to generate fake data, so anybody editing the code to their own purposes are unlikely to use this function at all